observability/src/main/java/io/grpc/observability/logging/CloudLoggingHandler.java

```diff
@@ -0,0 +1,190 @@
+    this(DEFAULT_LOG_LEVEL, null, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   */
+  public CloudLoggingHandler(Level level) {
+    this(level, null, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   * @param logName the name of the log to which log entries are written
+   */
+  public CloudLoggingHandler(Level level, String logName) {
+    this(level, logName, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   * @param logName the name of the log to which log entries are written
+   * @param overrideProjectId the value of cloud project id to which logs are sent to by the custom
+   *     logger
+   */
+  public CloudLoggingHandler(Level level, String logName, String overrideProjectId) {
+    baseLevel = level.equals(DEFAULT_LOG_LEVEL) ? Level.FINEST : level;
+    setLevel(baseLevel);
+
+    cloudLogName = logName != null ? logName : DEFAULT_LOG_NAME;
+
+    try {
+      // TODO(dnvindhya) read the value from config instead of taking it as an argument
+      String projectId = overrideProjectId != null ? overrideProjectId : null;
+
+      if (!Strings.isNullOrEmpty(projectId)) {
+        loggingOptions = LoggingOptions.newBuilder().setProjectId(overrideProjectId).build();
+      } else {
+        loggingOptions = LoggingOptions.getDefaultInstance();
+      }
+
+      loggingClient = loggingOptions.getService();
+    } catch (Exception ex) {
+      reportError("Error in initializing Cloud Logging", ex, ErrorManager.OPEN_FAILURE);
+      throw ex;
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (!(record instanceof LogRecordExtension)) {
+      throw new IllegalArgumentException("Expected record of type LogRecordExtension");
+    }
+
+    Level logLevel = record.getLevel();
+    GrpcLogRecord protoRecord = ((LogRecordExtension) record).getGrpcLogRecord();
+
+    writeLog(protoRecord, logLevel);
+  }
+
+  private void writeLog(GrpcLogRecord logProto, Level logLevel) {
+    try {
+      Severity cloudLogLevel = getCloudLoggingLevel(logLevel);
+      Map<String, Object> mapPayload = protoToMapConverter(logProto);
+
+      LogEntry grpcLogEntry =
+          LogEntry.newBuilder(JsonPayload.of(mapPayload))
+              .setSeverity(cloudLogLevel)
+              .setLogName(cloudLogName)
+              .setResource(MonitoredResource.newBuilder("global").build())
+              .build();
+
+      loggingClient.write(Collections.singleton(grpcLogEntry));
+    } catch (Exception ex) {
+      getErrorManager().error("Error in writing logs to cloud logging", ex,
```

Comment:
Couple of comments:

- why call `getErrorManager().error` here but `reportError()` on line 107? Can we be consistent?
- I don't know if we using the JUL Handler's default error reporting facility for our internal errors. For us JUL is just an abstraction to connect our producer with Cloud Logging so I don't see any reason to use JUL Handler here. Should we just allow the exception to propagate or use another mechanism to log these errors?